pkg/graph/graph.go

```diff
@@ -55,6 +56,9 @@ func (n ConfigNode) ID() int64 {
 
 // DOTID returns the node's identifier when printed to a DOT file. For readability of files this is the coordinate.Coordinate of the Config, instead of the node's ID integer.
 func (n ConfigNode) DOTID() string {
+	if n.DOTEncoding != "" {
+		return n.DOTEncoding
+	}
 	return n.Config.Coordinate.String()
```

Comment:
Just an observation: if using the "string" representation, the encoding is done late, here in this function, where as for JSON it is done much earlier, when building the graph. Is there a reason for this? All things being equal, would it be better to aways do the encoding at the same point in time?

Reply:
Personal preference would be to do it as late as possible. 
Based on your suggestion to allow the definition of the full encoding logic outside, rather than just flipping a bool option, I originally added an encoding function to the Node struct - but the realized that makes it non-comparable